tests/unit/utils/test_validations.py

```diff
@@ -0,0 +1,498 @@
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This variable should be a float value but is +-infinity, check the following variable:"
+    )
+    assert result[0].property_name == "float_test_variable"
+
+
+def test_validate_float_type_neg_infinity_incorrect():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting an float,
+    but it was provided with a float("-inf") instead.
+    """
+
+    test_instance = TestClass(
+        str_test_variable="test", int_test_variable=5, float_test_variable=float("-inf")
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This variable should be a float value but is +-infinity, check the following variable:"
+    )
+    assert result[0].property_name == "float_test_variable"
+
+
+def test_validate_float_type_correct():
+    """This test will check if the class with the correct argument types will pass the validation"""
+
+    test_instance = TestClass(
+        str_test_variable="test", int_test_variable=5, float_test_variable=3.14
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 0
+
+
+def test_validate_file_exists():
+    """
+    This test will check if the class with the correct argument types will pass the validation
+    """
+    test_instance = TestClass(
+        str_test_variable="test",
+        int_test_variable=5,
+        float_test_variable=3.14,
+        path_test_variable=_test_case_base_dir / "csv_file.csv",
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 0
+
+
+def test_validate_file_not_exists():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting a file that exists,
+    but it was provided with a file that does not exist instead.
+    """
+    test_instance = TestClass(
+        path_test_variable=_test_case_base_dir / "not_a_csv_file.csv",
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message == "This file does not exist, check the following variable:"
+    )
+    assert result[0].property_name == "path_test_variable"
+
+
+def test_validate_file_not_a_path():
+    """
+    This test will validate if the class has been instanciated with a variable that was expecting a valid file path,
+    but it was provided with an invalid file path instead.
+    """
+    test_instance = TestClass(
+        path_test_variable="test",
+    )
+
+    result = test_instance.validate()
+
+    assert len(result) == 1
+    assert (
+        result[0].message
+        == "This is not a valid file path, check the following variable:"
+    )
+    assert result[0].property_name == "path_test_variable"
+
+
+def test_validate_datetime_correct():
```

Comment:
Can you add docstring comments against all of the remaining tests?

Reply:
Docstring comments added for all remaining tests